### DISCUSSION TITLE
Unfold the types in code generator

### DISCUSSION THREAD

**Discussion by FlatMapIO** [chakra-ui/panda#2081] [Ideas]
Due to the extremely complex inference expressions on which the panda CSS properties depend, most of the time, the props autocompletion is only half-functional. Since panda relies on static code generation, I believe it is necessary to generate flat and explicit types. This will enable the IDE to work smoothly.

**Comment by astahmer** [comment]
can you make a minimal reproduction of your issue ?

**Comment by FlatMapIO** [comment]
I'm having a hard time creating a stable reproduction of this, essentially it's not a direct problem with panda, but rather the complexity of the panda types that makes IDE support for typescript a bit confusing and slow, and I've recorded a video of it.  I recorded a video. Taking colors as an example, the first time I completed the color list it was only the name, and I had to backspace->trigger autocomplete and some other confusing actions to find "all" symbols.

https://github.com/chakra-ui/panda/assets/1720349/b42c107c-eb09-4fa7-862d-34420f5611b3

**Reply by anubra266** [reply]
@FlatMapIO Try disabing the vscode extension and try again. This seems like an issue with the panda vscode extension.
